clients/horizonclient: Add SEP0029 memo required check. (#2396)
stellar/go: clients/horizonclient/CHANGELOG.md

## Unreleased

### Add 

- Add [SEP0029](https://github.com/stellar/stellar-protocol/blob/master/ecosystem/sep-0029.md) memo required check in `client.SubmitTransaction`.
    If any of the operations included in `client.SubmitTransaction` is of type
    `payment`, `pathPaymentStrictReceive`, `pathPaymentStrictSend`, or

>     `mergeAccount`, then the SDK will load the destination account and check if

Reviewer: >load the destination account

If I was reading this without context I wouldn't expect this to mean that this is a call out to Horizon, but I think it's important we call that out. Could we clarify that, maybe with the change below?
```suggestion
    `mergeAccount`, then the SDK will load the destination account from Horizon and check if
```

    `config.memo_required` is set to `1` as defined in SEP0029.

>     This behavior is the default. You can pass skip the check providing the

Reviewer 2: ```suggestion
    This behavior is the default. For performance reasons, you may choose to skip the check by providing the
```

    optional parameter  `horizonclient.SkipMemoRequiredCheck` to
    `client.SubmitTransaction`:

	```
		client.SubmitTransaction(tx, horizonclient.SkipMemoRequiredCheck)
	```

> 	The check is also skipped if the transaction includes a memo.

Reviewer 2: ```suggestion
	Additionally, the check will be skipped automatically if the transaction includes a memo.
```
